src/targets/cpu/fuse_ops.cpp

```diff
@@ -0,0 +1,110 @@
+#include <migraphx/cpu/fuse_ops.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/operation.hpp>
+#include <migraphx/value.hpp>
+#include <migraphx/matcher.hpp>
+#include <migraphx/context.hpp>
+#include <migraphx/env.hpp>
+#include <migraphx/cpu/context.hpp>
+#include <migraphx/dead_code_elimination.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace cpu {
+
+MIGRAPHX_DECLARE_ENV_VAR(MIGRAPHX_DISABLE_DNNL_POST_OPS_WORKAROUND);
+
+MIGRAPHX_PRED_MATCHER(has_post_ops, instruction_ref ins)
+{
+    auto v = ins->get_operator().to_value();
+    return v.contains("post_ops");
+}
+
+MIGRAPHX_PRED_MATCHER(without_post_ops, instruction_ref ins)
+{
+    auto v = ins->get_operator().to_value();
+    return v.contains("post_ops") and v["post_ops"].empty();
+}
+
+bool workaround_dnnl_broken_post_ops(const operation& op, const operation& post_op)
+{
+    if(contains({"dnnl::dot", "dnnl::convolution"}, op.name()))
```

Comment:
so all convolutions and gemm fusions are broken currently?

Reply:
They most likely need to use the 3 argument form. When using an 'add' post_op for convolution or dot it does work correctly but it usually falls back on a ref implementation which is incredibly slow. Other post_ops don't seem to work consistently.

This restricts it to only use post_ops when it is binary+eltwise. It does work in some other scenarios but its hard to characterize when it works and when it doesn't.

Reply:
I am not sure that we should call the function `workaround_dnnl_broken_post_ops` if it's just op+eltwise. Maybe something like `check_op_eltwise`? Within the function, we could also include a comment about why these fusions aren't supported currently.

Reply:
It is a workaround. dnnl shouldn't return an algorithm when the set of ops are not supported. I would prefer to keep the name whereas `check_op_eltwise` name would not be accurate if we refine this workaround in the future.